src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -626,13 +626,29 @@ def __init__(
             "file": file,
         }
 
+    def _close_hdf5_file(self):
+        has_hdf5_backend = hasattr(self, "_file")
+        if has_hdf5_backend:
+            import h5py
+
+            main_file_id = self._file.id
+            open_object_ids_main = h5py.h5f.get_obj_ids(main_file_id, types=h5py.h5f.OBJ_ALL)
+            for object_id in open_object_ids_main:
+                object_name = h5py.h5i.get_name(object_id).decode("utf-8")
+                try:
+                    object_id.close()
+                except:
+                    import warnings
+
+                    warnings.warn(f"Error closing object {object_name}")
```

Comment:
since this is the same function as the `Sorting` object, why not making a separate unique function?
```
def _close_hdf5_file(extraxctor):
    has_hdf5_backend = hasattr(extraxctor, "_file")
    ....
```

then you would just call it in the `__del__` with `_close_hdf5_file(self)`

Reply:
You are right that this repetition is unecessary but I wanted to use self, check out the latest commit. I implemented a mixing class that should hold the methods that need the state of the nwb extractor and are common to both recording and sorting.